src/Language/PureScript/Bundle.hs

```diff
@@ -202,24 +207,30 @@ withDeps (Module modulePath fn es) = Module modulePath fn (map expandDeps es)
       expand (ty, nm, n1, _) = (ty, nm, n1, ordNub (dependencies modulePath n1))
   expandDeps other = other
 
-  dependencies :: ModuleIdentifier -> JSExpression -> [(ModuleIdentifier, String)]
-  dependencies m = everything (++) (mkQ [] toReference)
+  dependencies :: ModuleIdentifier -> JSExpression -> [Key]
+  dependencies m = everythingWithContext boundNames (++) (mkQ (const [] &&& id) toReference)
     where
-    toReference :: JSExpression -> [(ModuleIdentifier, String)]
-    toReference (JSMemberDot mn _ nm)
+    toReference :: JSExpression -> [String] -> ([Key], [String])
+    toReference (JSMemberDot mn _ nm) bn
       | JSIdentifier _ mn' <- mn
       , JSIdentifier _ nm' <- nm
       , Just mid <- lookup mn' imports
-      = [(mid, nm')]
-    toReference (JSMemberSquare mn _ nm _)
+      = ([(mid, nm', Public)], bn)
+    toReference (JSMemberSquare mn _ nm _) bn
       | JSIdentifier _ mn' <- mn
       , Just nm' <- fromStringLiteral nm
       , Just mid <- lookup mn' imports
-      = [(mid, nm')]
-    toReference (JSIdentifier _ nm)
-      | nm `elem` boundNames
-      = [(m, nm)]
-    toReference _ = []
+      = ([(mid, nm', Public)], bn)
+    toReference (JSIdentifier _ nm) bn
+      | nm `elem` bn
```

Comment:
Will comment; this guard is half of the point of this PR, so it needs to be immediately understandable!

How would this read to you?
> `-- only add a dependency if this name is still in the list of names bound to the module level (i.e., hasn't been shadowed by a function parameter)`

Or would it be clearer if `boundNames`/`bn` got more descriptive names—`moduleBoundNames`/`namesStillBoundToModule`, perhaps? The style in use here seems to eschew overly verbose variable names, which is why I went for the terse option, but clearly that was at the expense of immediate comprehension.

Reply:
The issue was not me not knowing what `bn` meant, but rather me (incorrectly) thinking it could be done away with now that we are tracking visibility in the graph. That comment would be a great addition, thanks!